beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -54,3 +60,142 @@ func (vs *Server) getBellatrixBeaconBlock(ctx context.Context, req *ethpb.BlockR
 	blk.StateRoot = stateRoot
 	return blk, nil
 }
+
+// This function retrieves the payload header given the slot number and the validator index.
+// It's a no-op if the latest head block is not versioned bellatrix.
+func (vs *Server) getPayloadHeader(ctx context.Context, slot types.Slot, idx types.ValidatorIndex) (*ethpb.ExecutionPayloadHeader, error) {
+	if err := vs.BlockBuilder.Status(); err != nil {
+		return nil, err
+	}
+	b, err := vs.HeadFetcher.HeadBlock(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if blocks.IsPreBellatrixVersion(b.Version()) {
+		return nil, nil
+	}
+	h, err := b.Block().Body().ExecutionPayload()
+	if err != nil {
+		return nil, err
+	}
+	pk, err := vs.HeadFetcher.HeadValidatorIndexToPublicKey(ctx, idx)
+	if err != nil {
+		return nil, err
+	}
+	bid, err := vs.BlockBuilder.GetHeader(ctx, slot, bytesutil.ToBytes32(h.BlockHash), pk)
+	if err != nil {
+		return nil, err
+	}
+	return bid.Message.Header, nil
+}
+
+// This function constructs the builder block given the input altair block and the header. It returns a generic beacon block for signing
+func (vs *Server) buildHeaderBlock(ctx context.Context, b *ethpb.BeaconBlockAltair, h *ethpb.ExecutionPayloadHeader) (*ethpb.GenericBeaconBlock, error) {
```

Comment:
does this function and the get builder block need nil checks for b and h parameters since these are APIs